assets/js/src/core/modules/data-object/actions/add-object/use-add-object.tsx

```diff
@@ -0,0 +1,149 @@
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import { type ItemType } from '@Pimcore/components/dropdown/dropdown'
+import { type TreeNodeProps } from '@Pimcore/components/element-tree/node/tree-node'
+import { Icon } from '@Pimcore/components/icon/icon'
+import { useFormModal } from '@Pimcore/components/modal/form-modal/hooks/use-form-modal'
+import { type ClassDefinitionListItem } from '@Pimcore/modules/class-definition/class-definition-slice.gen'
+import { useClassDefinitions } from '@Pimcore/modules/class-definition/hooks/use-class-definitions'
+import _ from 'lodash'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDataObjectAddMutation } from '../../data-object-api-slice.gen'
+import { useAppDispatch } from '@Pimcore/app/store'
+import { refreshNodeChildren } from '@Pimcore/components/element-tree/element-tree-slice'
+import trackError, { ApiError, GeneralError } from '@Pimcore/modules/app/error-handler'
+
+interface UseAddObjectHookReturn {
+  addObjectTreeContextMenuItem: (node: TreeNodeProps) => ItemType
+}
+
+export const useAddObject = (): UseAddObjectHookReturn => {
+  const { t } = useTranslation()
+  const classDefinitions = useClassDefinitions()
+  const modal = useFormModal()
+  const [addDataObjectMutation] = useDataObjectAddMutation()
+  const dispatch = useAppDispatch()
+
+  const getClassEntries = (node: TreeNodeProps): ItemType[] => {
+    let classHirachy: ItemType[] = []
+
+    const structuredClassDefinitions = [...classDefinitions]
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .reduce<Record<string, ClassDefinitionListItem[]>>((acc, classDefinition) => {
+      const groupName = _.isEmpty(classDefinition.group)
+        ? 'undefined'
+        : classDefinition.group
+
+      if (acc[groupName] === undefined) {
+        acc[groupName] = []
+      }
+
+      acc[groupName].push(classDefinition)
+
+      return acc
+    }, {})
+
+    if (structuredClassDefinitions.undefined !== undefined) {
+      classHirachy = structuredClassDefinitions.undefined.map(classDefinition => getDataObjectEntry(classDefinition, node))
+    }
+
+    for (const [group, classDefinitions] of Object.entries(structuredClassDefinitions)) {
+      if (group !== 'undefined') {
+        classHirachy.push({
+          label: group,
+          key: 'add-object-group-' + group,
+          icon: <Icon value={ 'folder' } />,
+          children: classDefinitions.map(classDefinition => getDataObjectEntry(classDefinition, node))
+        })
+      }
+    }
+
+    return classHirachy
+  }
+
+  const getDataObjectEntry = (classDefinition: ClassDefinitionListItem, node: TreeNodeProps): ItemType => {
+    return {
+      label: classDefinition.name,
+      key: classDefinition.id,
+      icon: <Icon { ...classDefinition.icon } />,
```

Comment:
The default icon does not work. I only see icons for class definitions which do define their own icons.

![image](https://github.com/user-attachments/assets/e2e0aa9c-b93a-435f-a134-277435820a2c)

Reply:
https://github.com/pimcore/studio-backend-bundle/issues/893